docs/website/root/mithril/threat-model.md

```diff
@@ -0,0 +1,293 @@
+
+#### Block diffusion
+
+Block diffusion process ensures the timely diffusion of blocks, both those produced "locally" and those received from upstream peers
+* **availability**: Yes (Not being to diffuse blocks harms a BP's SPO economic viability, and can also harm their ability to create new blocks)
+* **confidentiality**: No (block diffusion happens in the open)
+* **integrity**: Yes (partially? not sure what integrity means here)
+
+#### Block production
+
+Block production is the process of "minting" new blocks by block producers, driven by Stake-based random lottery.
+
+The mithril-signer necessarily runs on the same host than a BP because it needs access to the KES signing key
+
+* **availability**: Yes (BP is critical for SPOs revenue, and preventing a BP from producing blocks can harm SPOs capabilities to operate)
+* **integrity**: Yes (incorrect or invalid data can hamper BP capabilities)
+* **confidentiality**: Yes? (BP schedule is private information for the BP, leaking it could provide adversaries advance knowledge of schedules and lead to **grinding attacks** to attempt to manipulate nonce in the disfavour of a BP ??)
+
+#### Cardano Chain Database
+
+A cardano-node maintains an on-disk database consisting of the chain's history. This database is updated by the node when new blocks are diffused through the network, or minted, and also contains a cache of the ledger state. 
+
+Mithril signer needs needs access to _trusted_ and _up-to-date_ Chain database in order to be able to sign snapshots. 
+
+* **availability**: Yes
+* **integrity**: Yes
+* **Confidentiality**: No (Data is public and replicated) 
+
+#### Cardano Ledger state
+
+Access to an accurate ledger state is needed by mithril signer to retrieve reliable _Stake distribution_. This access is currently done through a local connection (direct w/ Pallas or indirect with cardano-cli) to a trusted cardano-node
+
+* **availability**: Yes (without SD, signer cannot register keys nor validly use other signers' keys)
+* **integrity**: Yes (same, inaccurate SD will make key registration and signing process invalid)
+* **confidentiality**: No
+
+#### Mithril signing keys
+
+SPOs register their Mithril keys every epoch to be able to sign snapshots. An attacker could impersonate the SPO and sign invalid snapshots if they got hold of those keys
+Signing keys are currently stored on-disk (?) 
+
+* **confidentiality**: Yes 
+* **integrity**: Yes (invalid key is useless obviously)
+* **availability**: Yes (Signer needs Key at every signing round, unavailability will lead to inability to sign)
+
+#### Mithril signing keys registration
+
+Mithril signer needs to register new key every epoch with aggregator (and ultimately other signers)
+
+* **confidentiality**: Yes (access to a signer's key will allow an attacker to impersonate a signer for the duration of the epoch)
```

Comment:
```suggestion
* **confidentiality**: No (only verification keys and proofs of possession are used in the process) 
```

Reply:
The registration takes vk and pop; however, vk is generated by the signing key. In this case, we may make clear that the confidentiality of vk and pop is not necessary, but the signing key is.

Reply:
We should keep the `(access to a signer's key will allow an attacker to impersonate a signer for the duration of the epoch)` as an explanation why confidentiality of the signing key itself is important (one section above).

Reply:
IMO we should probably make a clear separation between the two sections:
- Mithril signing keys (private information, must be kept secret)
- Mithril verification keys (public information, must be registered with the other signers)
WDYT?

Reply:
Done